src/internal.cpp

```diff
@@ -292,13 +308,17 @@ void Internal::DataHandler(sFrameOfMocapData* data, void* pUserData, Internal &i
     internal.UnlabeledCount = 0; 
 }
 
+void Internal::initROSmsgHeader(std_msgs::Header &msgHeader, Internal &internal)
+{
+    msgHeader.frame_id = internal.rosparam.mocap_base_frame;
+    msgHeader.stamp = timeStampLatestFrame;
+}
+
 void Internal::PubRigidbodyPose(sRigidBodyData &data, Internal &internal)
 {
     // Creating a msg to put data related to the rigid body and 
     geometry_msgs::PoseStamped msgRigidBodyPose;
-    msgRigidBodyPose.header.frame_id = "world";
-
-    msgRigidBodyPose.header.stamp = ros::Time::now();
+    initROSmsgHeader(msgRigidBodyPose.header, internal);
```

Comment:
This looks good. I have one question, 
It is not a good idea to use `initROSmsgHeader(msgRigidBodyPose.header, internal);` for each body and markers. Because I think it might cause major time stamp difference when there are multiple bodies and code will loop through all the number of frames for publishing.
I will modify it a bit.

Reply:
didn't follow. all the bodies in one frame should have the same time frame

Reply:
suppose we have a huge arena and we have multiple rigid bodies in parallel with multiple single marker tracking, so what your implementation is doing, is assigning all the same time stamp. But publishing is not done at the same time. Publishing is FIFO so all the body and markers will have slightly different time stamp (difference in nano seconds)

Reply:
Yes that's where I am bit confused.
What makes more sense to you, to consider the time of arrival of data in queue or time of exit minus the communication delay between the systems?

Reply:
Having a header timestamp pointing to as precise of a time of exact occurrence helps for controls tuning, debugging etc

Reply:
I see. I will keep it optional and let users choose one of the model for the timestamp in the launch file.